lib/resources/shim-head.js

```diff
@@ -2,6 +2,25 @@
 ((win) => {
   win.ELECTRON = true;
 
+  if (process.env.ELECTRON_PROTOCOL_SERVE_INDEX && window.location.protocol !== 'file:') {
```

Comment:
I don't want to check for `serve:` because `electron-protocol-serve` allows the user to customize the protocol they use. `serve:` is just the default. I suppose I could jam that into yet another environment variable, but I don't think it's necessary.

This shim is the very first piece of script that runs on the page (unless the user does something really inadvisable), and only runs once, so at this point we can be confident that `window.location` is the URL of `index.html`. So there's no risk of it having changed so some HTTP URL or something.